clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3619,4 +3619,14 @@ protected boolean isTransientRecordBufferUsed() {
   protected void setPartitionConsumptionState(int partition, PartitionConsumptionState pcs) {
     partitionConsumptionStateMap.put(partition, pcs);
   }
+
+  public void blockOnDrainerCapacity() {
+    while (storeBufferService.getTotalRemainingMemory() < 1000) {
```

Comment:
I agree... would be great to avoid sleeping, if possible. I've been meaning to get rid of sleeps (e.g., did it in 6db7c05f8be998020f865cceec44640ffdfd8d08) and I hope we continue getting rid of them, rather than add more.

Reply:
will do